realsense_camera/src/base_nodelet.cpp

```diff
@@ -275,34 +276,63 @@ namespace realsense_camera
   {
     // print list of detected cameras
     std::vector<int> camera_type_index;
-    std::string detected_camera_msg = " - Detected the following cameras:";
+
+    // map camera names to validated firmware
+    std::map<std::string, std::string> camera_name_to_validated_firmware =
+    {
+          {"Intel RealSense R200", R200_CAMERA_FW_VERSION},
+          {"Intel RealSense F200", F200_CAMERA_FW_VERSION},
+          {"Intel RealSense SR300", SR300_CAMERA_FW_VERSION},
+          {"Intel RealSense ZR300", ZR300_CAMERA_FW_VERSION},
+          {"Intel RealSense ZR300_ADAPTER", ZR300_ADAPTER_FW_VERSION},
+          {"Intel RealSense ZR300_MOTION_MODULE", ZR300_MOTION_MODULE_FW_VERSION}
+    };
+
     for (int i = 0; i < num_of_cameras; i++)
     {
+      std::string detected_camera_msg = " - Detected the following camera:";
```

Comment:
Does this not result in printing the message for each camera found instead of once for all cameras?